src/commonMain/kotlin/dev/snipme/highlights/Highlights.kt

```diff
@@ -10,13 +11,22 @@ import dev.snipme.highlights.model.PhraseLocation
 import dev.snipme.highlights.model.SyntaxLanguage
 import dev.snipme.highlights.model.SyntaxTheme
 import dev.snipme.highlights.model.SyntaxThemes
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.ensureActive
+import kotlinx.coroutines.launch
+import kotlin.coroutines.cancellation.CancellationException
 
 class Highlights private constructor(
     private var code: String,
     private val language: SyntaxLanguage,
     private val theme: SyntaxTheme,
     private var emphasisLocations: List<PhraseLocation>
 ) {
+    private var analysisJob: Job? = null
```

Comment:
This is still not thread safe because the job is not handled atomically. If you want compatibility with Swift, consider creating a separate Flow interface that has a listener API